rosbag2_snapshot/src/snapshotter.cpp

```diff
@@ -738,6 +761,26 @@ bool Snapshotter::writeTopic(
   return true;
 }
 
+template<typename MsgType>
+bool Snapshotter::isMsgInsideInterval(
+  const MsgType& msg,
+  const rosbag2_snapshot_msgs::srv::TriggerSnapshot::Request::SharedPtr& req, // The request object, assuming a common type here
+  MessageQueue::queue_t::const_iterator msg_it, // Iterator to the current message, assuming a common type
+  const TopicDetails& topic_details) 
+{
+  // Calculate time difference
+  double nsec_diff = msg.header.stamp.nanosec - req->msg_timestamp.nanosec;
+  double diff_sec = std::abs(msg.header.stamp.sec - req->msg_timestamp.sec) + std::abs(nsec_diff * 1e-9);
```

Comment:
I think this logic is wrong. Say the request was made at t=0.99999999 seconds and the message is at t=1.0000 seconds. `nsec_diff` would be 9999... nanoseconds, and `diff_sec` would be 1 second. Summing in abs value you would get 1.9999 seconds, when this is actually 1 nanosecond apart

Reply:
troll 😮‍💨 fixed, thanks!

Reply:
Actually, this logic was completely removed in the latest changes. Not required anymore